libcalico-go/lib/clientv3/interface.go

```diff
@@ -21,53 +21,128 @@ import (
 )
 
 type Interface interface {
+	NodesClient
```

Comment:
I split out these smaller interfaces to make mocking easier.  Then I refactored the code and I didn't need to mock the NodeClient any more.  Nevertheless, it seemed like a good idea to make mocking easier in future...